movementlabsxyz/movement
# Feat: E2E test test_movement_client_should_successfully_call_lock_and_complete

# Summary
- Implement `test_movement_client_should_successfully_call_lock_and_complete` to almost fully satisfy issue #313 

# Changelog

- Modify Move functions so they are entry, which meant they could not return a value. Modify Move unit tests accordingly, did a little clean up of arg names (replaced `originator` with `initiator` throughout).
- Fix publishing step so `MovementClient` private key is used to generate profile in `movement init` and consequently used for publishing
- Modify `send_and_confirm_aptos_transaction` utility function to use current sequence number (to fix "sequence number too old" error)
- Fix serialization issues for function args

# Testing

To test, run: 

```
rust_backtrace=1 cargo test --test eth_movement test_movement_client_should_successfully_call_lock -- --nocapture --test-threads=1
```
You can see `Response:` for each test in the output.

If a test fails, then to restart the local network use `ps aux | grep movement` to find the running process and then `kill -9 <process-id>` to kill it.

# Outstanding issues
- Still need to write test for aborting bridge transfer.
- I left the test functions in the Move module. Once this PR is reviewed, I can remove them. Happy to remove them anytime, just wanted them there so you could follow the debugging logic.
